specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/schema/kafka.schema.patch.json

```diff
@@ -82,6 +82,52 @@
                                     }
                                 }
                             },
+                            "catalog":
+                            {
+                                "title": "Catalog",
+                                "type": "object",
+                                "patternProperties":
+                                {
+                                    "^[a-zA-Z]+[a-zA-Z0-9\\._\\-]*$":
+                                    {
+                                        "topic":
+                                        {
+                                            "type": "string"
+                                        },
+                                        "key":
+                                        {
+                                            "strategy":
+                                            {
+                                                "type": "string"
+                                            },
+                                            "version":
+                                            {
+                                                "type": "string"
+                                            },
+                                            "id":
+                                            {
+                                                "type": "integer"
+                                            }
```

Comment:
Require `strategy` + (optional) `version` to resolve the `id` dynamically or fixed `id`.

Reply:
We need some validation here, since it only makes sense to specify `id` by itself, and not alongside `strategy` or `version`.